src/experiments/misc/Raja_experiment.ml

```diff
@@ -35,11 +35,17 @@ let ranges_of_path (path : Fpath.t) : Function_range.ranges =
   match Hashtbl.find_opt cache path with
   | Some x -> x
   | None ->
-      (* alt: could use just_parse_with_lang and get the lang
-       * from the languages: field in the rule corresponding to
-       * the rule_id of the match
-       *)
-      let ast = Parse_target.parse_program (Fpath.to_string path) in
+      let ast =
+        (* parse_program may raise the "hd" exn when we can't infer
+         * the language from the filename.
+         * alt: we could use just_parse_with_lang and get the language
```

Comment:
I definitely like the alt suggestion better. This is an experiment so ok with this for now, but maybe change the "alt" to "TODO"